src/tests/verify_authfile.sh

```diff
@@ -2,4 +2,4 @@
 
 cd "$(dirname "$0")/../../"
 export PYTHONPATH=src/:$PYTHONPATH
-exec ./bin/osg-authfile > /dev/null
+exec ./bin/osg-authfile > /dev/null 2>&1
```

Comment:
So, you want to hide a traceback when there are errors?

Reply:
Yeah, AIUI, these tracebacks would be logged but since we're running the functions manually, they get dumped to `stderr`. The functions themselves handle these particular exceptions and so they exit zero, we just want to throw away their out (I forget why, @matyasselmeci ?)  and err because GitHub Actions apparently look for these tracebacks and "annotate" the test run with an error but otherwise lets the test pass because the script exist zero. See https://github.com/brianhlin/topology/actions/runs/1144133515. Exceptions that we aren't catching should still cause the script to exit non-zero.

**tl;dr** yeah, we want to throw these tracebacks away since we catch them otherwise they just lead to confusing GHA results

Reply:
Well i admit i am confused.

But let's see.  IIUC:

- there are uncaught exceptions
- these produce tracebacks on stderr
- GHA uses stderr output to annotate results
- these GHA runs still pass because (for some reason the script) still exits `0`
- we throw out stderr entirely to avoid the confusing GHA annotations with such "passing" jobs
- uncaught exceptions should still cause the script to exit non-zero

So, for this guy,

> ```Error:     raise LDAPPasswordIsMandatoryError('password is mandatory in simple bind')```

is this an exception we catch & handle, but print the traceback anyway, or is it an uncaught exception that _should_ cause the script to exit non-zero, but doesn't?

Reply:
Oops, I broke off the comments into the main PR thread. See that for the explanation